lld/MachO/SyntheticSections.cpp

```diff
@@ -2084,7 +2084,7 @@ void ObjCMethListSection::writeTo(uint8_t *bufStart) const {
     uint32_t writtenSize = writeRelativeMethodList(isec, buf);
     buf += writtenSize;
   }
-  assert(buf - bufStart == sectionSize &&
+  assert(buf - bufStart == long(sectionSize) &&
```

Comment:
```suggestion
  assert(buf - bufStart == std::ptrdiff_t(sectionSize) &&
```
Seems more correct? Same for the `assert` above.

Reply:
Okay let me try the above and get back

Reply:
Thanks for the review @carlocab 
I have applied the above suggestion. It does make sense to use it in place of `long`